### ISSUE TITLE
Sometimes fails to get verbs from dropbox

### ISSUE THREAD

**Issue by srandall52** [Jeinzi/thunar-dropbox#2]
At line 260 in tdp-provider.c, the loop to retrieve dropbox options terminates on G_IO_STATUS_AGAIN. Elsewhere this status causes the operation to be retried. And, in fact, if I simply delete these three lines, I am able to get a proper menu when otherwise I don't.

**Comment by Jeinzi** [comment]
Thank you for your bug report, I'll have a look!

**Comment by Jeinzi** [comment]
Do you have any way of provoking G_IO_STATUS_AGAIN?

**Comment by srandall52** [comment]
Yes. It's been intermittent for years, but happens consistently now. I assume the status means that Dropbox isn't responding fast enough, possibly because my computer is old and slow. Or my network connection.

I might add that it works fine in other directories, where there is only one menu entry (Move to Dropbox). It's only in the Dropbox directory itself that there's a problem.

**Comment by wolfgang-noichl** [comment]
For me, this prevents any Dropbox menus being displayed. See above for a fix. There is a noticeable delay in any actions the Dropbox folder, so in the long run, it might make sense to add these menu items without querying Dropbox at all, see [here](http://api.xfce.m8t.in/xfce-4.10/thunarx-1.4.0/ThunarxMenuProvider.html):

> Extensions that implement this interface should do its best to respond as fast as possible to method invokations by the file manager.

There is a "share" button that changes to "manage access" in case the folder is shared already. Other than that, I think it would suffice to figure out whether a particular folder is inside Dropbox or not.

**Comment by srandall52** [comment]
As observed by @wolfgang-noichl, the "fix" comes at a cost. I see that in folders like Computer and Trash, Dropbox returns G_IO_STATUS_AGAIN, even though no amount of retrying can ever change anything. That seems like a Dropbox bug, but what do I know. Anyway, it causes Thunar to hang.

One can avoid the hang by just not going into those troublesome folders, and for me that was better than losing the menu. But it's certainly not how the plugin should work by default.

**Comment by wolfgang-noichl** [comment]
I think it makes sense to leave this bug open, since the underlying issue is clearly still there :)

Having read [this](https://mail.gnome.org/archives/gtk-app-devel-list/2005-October/msg00350.html), I think the main problem is that there is a "blocking infinite loop" in this plugin while there _should_ be something that just asynchronously tries again every now and then.

A ugly, yet easy workaround might be trying e.g. three times with 0.5 seconds in between, and then giving up.

**Comment by srandall52** [comment]
I'm OK with leaving this open. As you say, there is still an issue. I don't see any good resolution, though.

I might add that I have a new laptop now and can no longer reproduce this issue. Otherwise I'd be tempted to try the ugly solution myself.
